libnavigation-core/src/main/java/com/mapbox/navigation/core/preview/RoutesPreview.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.navigation.core.preview
+
+import androidx.annotation.StringDef
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteMetadata
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class RoutesPreview(
+    val primaryRoute: NavigationRoute,
```

Comment:
```suggestion
    val selectedIndex: Int,
```

This is a non-mutable data object, so having an index into a list is nice and safe. 

We will also need to use the index. If it is an object we need to use an unsafe function like
```
 fun selectedIndex(): Int = originalRoutesList.indexOf(primaryRoute)
```